terraform/deployments/govuk-publishing-platform/signon_bearer_tokens.tf

```diff
@@ -31,140 +31,74 @@ locals {
+      app      = local.signon_app.publishing_api.name
+    }
+
+    cs_to_router_api = {
+      api_user = local.signon_api_user.content_store
+      app      = local.signon_app.router_api.name
+    }
+
+    dcs_to_pub_api = {
+      api_user = local.signon_api_user.draft_content_store
+      app      = local.signon_app.publishing_api.name
+    }
+
+    dcs_to_draft_router_api = {
+      api_user = local.signon_api_user.draft_content_store
+      app      = local.signon_app.draft_router_api.name
+    }
+
+    pub_to_pub_api = {
+      api_user = local.signon_api_user.publisher
+      app      = local.signon_app.publishing_api.name
+    }
+
+    pub_api_to_cs = {
+      api_user = local.signon_api_user.publishing_api
+      app      = local.signon_app.content_store.name
+    }
+
+    pub_api_to_dcs = {
+      api_user = local.signon_api_user.publishing_api
+      app      = local.signon_app.draft_content_store.name
+    }
+
+    pub_api_to_router_api = {
+      api_user = local.signon_api_user.publishing_api
+      app      = local.signon_app.router_api.name
+    }
+
+    frontend_to_pub_api = {
+      api_user = local.signon_api_user.frontend
+      app      = local.signon_app.publishing_api.name
+    }
+  }
 
-  additional_tags                 = local.additional_tags
-  api_user_email                  = local.signon_api_user.content_store
-  app_name                        = local.signon_app.router_api.name
-  environment                     = var.govuk_environment
-  name                            = "cs_to_router_api"
-  private_subnets                 = local.private_subnets
-  signon_admin_password_arn       = aws_secretsmanager_secret.signon_admin_password.arn
-  signon_host                     = module.signon.virtual_service_name
-  signon_lambda_security_group_id = aws_security_group.signon_lambda.id
-  workspace                       = local.workspace
+  generated_bearer_tokens = {
+    for user, user_email in local.signon_api_user :
+    user => [
+      for k, v in local.signon_bearer_tokens : module.signon_bearer_tokens[k].token_data
+      if v.api_user == user_email
+    ]
+  }
 }
 
-module "draft_content_store_to_draft_router_api_bearer_token" {
-  source = "../../modules/signon_bearer_token"
+module "signon_bearer_tokens" {
+  for_each = local.signon_bearer_tokens
+  source   = "../../modules/signon_bearer_token"
 
   additional_tags                 = local.additional_tags
-  api_user_email                  = local.signon_api_user.draft_content_store
-  app_name                        = local.signon_app.draft_router_api.name
+  api_user_email                  = each.value.api_user
+  app_name                        = each.value.app
+  name                            = each.key
   environment                     = var.govuk_environment
-  name                            = "dcs_to_draft_router_api"
   private_subnets                 = local.private_subnets
+  providers                       = { aws = aws, archive = archive }
```

Comment:
It's the `for_each` - you need to pass down the providers when using a `for_each` on the module. In other cases it's not needed since the child module inherits them automatically.

Reply:
Oh, how odd! Thanks :)